### ISSUE TITLE
Format on save not working

### ISSUE THREAD

**Issue by sashafklein** [prettier/prettier-atom#502]
(I've looked at three other similarly named tickets, and can't find a solution). 

The auto-format on save feature stopped working for me, although it doesn't throw any visible errors, and the indicator at the bottom of my Atom screen shows the green checkmark. Not sure what caused the change. Prettier still runs just fine (I have it run on commit using `husky` and `pretty-quick`).

My prettier config is in `package.json` (under a `prettier` key), and I have no `.prettierignore` file.

It even works when I run `Prettier: Format` explicitly (and follows the one Prettier config rule in the `package.json` file).

I tried uninstalling/reinstalling this package, but no luck.

Here's the output of `Prettier: Debug`:

```
Atom version: 1.37.0
prettier-atom version: 0.56.6
prettier: bundled
prettier version: 1.15.2
prettier-eslint version: 8.8.2
prettier-atom configuration: {
  "formatOnSaveOptions": {
    "isDisabledIfNotInPackageJson": true,
    "showInStatusBar": true,
    "enabled": true,
    "respectEslintignore": true,
    "excludedGlobs": [],
    "whitelistedGlobs": [],
    "isDisabledIfNoConfigFile": false,
    "ignoreNodeModules": true
  },
  "useEslint": true,
  "useStylelint": true,
  "prettierEslintOptions": {
    "prettierLast": false
  }
}
```

Thanks! Love the tool. Let me know if you need any more info.

**Comment by robwise** [comment]
Hey sorry this one slipped through the cracks, can you check the Atom console to see if there are any unreported errors happening? `cmd + option + i`

**Comment by sashafklein** [comment]
I'm sorry. I missed your response. But I've since switched to VSCode. Assuming this issue isn't happening for other people, I think you should close it, but if you want me to check that out, I can open up Atom and try that.

**Comment by robwise** [comment]
No worries, let me know if you ever do want to investigate further.

**Comment by bobisjan** [comment]
Hi @robwise,

I've the same issue, currently using `atom#1.41.0-nightly1`:

```
Atom version: 1.41.0-nightly1
prettier-atom version: 0.57.1
prettier: bundled
prettier version: 1.18.2
prettier-eslint version: 9.0.0
prettier-atom configuration: {
  "formatOnSaveOptions": {
    "isDisabledIfNoConfigFile": true,
    "isDisabledIfNotInPackageJson": true,
    "showInStatusBar": true,
    "enabled": true,
    "respectEslintignore": true,
    "excludedGlobs": [],
    "whitelistedGlobs": [],
    "ignoreNodeModules": true
  },
  "prettierOptions": {
    "singleQuote": true
  },
  "useEslint": false,
  "useStylelint": false,
  "prettierEslintOptions": {
    "prettierLast": false
  }
}
```

and in the console there is an error coming from https://github.com/prettier/prettier-atom/blob/master/src/linterInterface/index.js#L19:

```
prettier-atom attempted to set messages with linter package, but was unable. Messages: []

setMessages | @ | /Users/bobisjan/.ato…terface/index.js:22
-- | -- | --
  | clearLinterErrors | @ | /Users/bobisjan/.ato…terface/index.js:29
  | An.tap | @ | /Users/bobisjan/.ato…h/lodash.min.js:106
  | l | @ | /Users/bobisjan/.ato…sh/lodash.min.js:50
  | (anonymous) | @ | /Users/bobisjan/.ato…sh/lodash.min.js:49
  | attemptWithErrorNotification | @ | /Users/bobisjan/.ato…tOnSave/index.js:29
  | (anonymous) | @ | /Users/bobisjan/.ato…terface/index.js:64
  | asyncGeneratorStep | @ | /Users/bobisjan/.ato…yncToGenerator.js:3
  | _next | @ | /Users/bobisjan/.ato…ncToGenerator.js:25
  | (anonymous) | @ | /Users/bobisjan/.ato…ncToGenerator.js:32
  | (anonymous) | @ | /Users/bobisjan/.ato…ncToGenerator.js:21
  | attemptWithErrorNotification | @ | /Users/bobisjan/.ato…terface/index.js:76
  | safeFormatOnSaveIfAppropriate | @ | /Users/bobisjan/.ato…tOnSave/index.js:29
  | subscriptions.add.editor.getBuffer.onWillSave | @ | /Users/bobisjan/.ato…tom/dist/main.js:73
  | simpleDispatch | @ | <embedded>:11
  | (anonymous) | @ | <embedded>:11
  | emitAsync | @ | <embedded>:11
  | saveTo | @ | <embedded>:11
  | async function (async) |   |  
  | saveTo | @ | <embedded>:11
  | save | @ | <embedded>:11
  | save | @ | <embedded>:11
  | o.function.e.save.promisify | @ | <embedded>:11
  | promisify | @ | <embedded>:11
  | saveItem | @ | <embedded>:11
  | saveActiveItem | @ | <embedded>:11
  | saveActivePaneItem | @ | <embedded>:11
  | core:save | @ | <embedded>:11
  | handleCommandEvent | @ | <embedded>:11
  | KeymapManager.dispatchCommandEvent | @ | <embedded>:11
  | KeymapManager.handleKeyboardEvent | @ | <embedded>:11
  | handleDocumentKeyEvent | @ | <embedded>:11
```

Hope this helps ☺️.

**Comment by samselikoff** [comment]
I just updated this package and noticed the same thing. Downgrading to 0.56.6 fixed it for me.

Let me know how I can help!

**Comment by robwise** [comment]
I wonder if these are really both to the same issue then as the OP was on 56.6 when it happened?

Do you know why the indie delegate is null in your case? We require using either the `linter` package (or the atom nuclide thing that pretends its that) in order for us to display messages to you.

**Comment by amk221** [comment]
I'm experiencing this too, with `Only format if Prettier is found in your project's dependencies` switched on, prettier on save doesn't work.

```
Atom version: 1.39.1
prettier-atom version: 0.57.2
prettier: bundled
prettier version: 1.18.2
prettier-eslint version: 9.0.0
prettier-atom configuration: {
  "formatOnSaveOptions": {
    "enabled": true,
    "isDisabledIfNoConfigFile": true,
    "showInStatusBar": true,
    "isDisabledIfNotInPackageJson": true,
    "respectEslintignore": true,
    "excludedGlobs": [],
    "whitelistedGlobs": [],
    "ignoreNodeModules": true
  },
  "prettierOptions": {
    "singleQuote": true
  },
  "useEslint": true,
  "useStylelint": true,
  "prettierEslintOptions": {
    "prettierLast": true
  }
}
```

**Comment by bobisjan** [comment]
With updated Atom the format on save is working again 🎉.

```
Atom version: 1.41.0-nightly6
prettier-atom version: 0.57.2
prettier: bundled
prettier version: 1.18.2
prettier-eslint version: 9.0.0
prettier-atom configuration: {
  "formatOnSaveOptions": {
    "enabled": true,
    "isDisabledIfNoConfigFile": true,
    "isDisabledIfNotInPackageJson": true,
    "showInStatusBar": true,
    "respectEslintignore": true,
    "excludedGlobs": [],
    "whitelistedGlobs": [],
    "ignoreNodeModules": true
  },
  "prettierOptions": {
    "singleQuote": true
  },
  "useEslint": false,
  "useStylelint": false,
  "prettierEslintOptions": {
    "prettierLast": false
  }
}
```

**Comment by radfahrer** [comment]
I'm on Atom 1.40 and still experiencing this bug.

**Comment by robwise** [comment]
Do you have `linter` installed?

**Comment by thomasw** [comment]
> I'm experiencing this too, with Only format if Prettier is found in your project's dependencies switched on, prettier on save doesn't work. (prettier is in my project's devDependencies)

I just ran into this issue. Format on save wasn't working and there were no logs in the console.

In my case, I too had both the eslint integration enabled and "Only format if prettier is found in your project's dependencies" setting enabled. Manually reformatting worked just fine.

I didn't have prettier-eslint installed though. Once I added it to the project, everything worked as expected. I somehow missed the note in the settings panel saying what dependencies the plugin would look for if the eslint integration was toggled on. I know it says it very clearly there, but we might want to consider logging a warning whenever the plugin decides to do nothing for an otherwise supported file type.
